tests/nonbonded/test_nonbonded_interaction_group.py

```diff
@@ -219,3 +227,53 @@ def test_ixngroups(conf):
         ref_delta = ref_allpairs(conf_prime) - ref_allpairs(conf)
         test_delta = test_ixngroups(conf_prime) - test_ixngroups(conf)
         np.testing.assert_allclose(ref_delta, test_delta, rtol=rtol, atol=atol)
+
+
+@pytest.mark.parametrize("beta", [2.0])
+@pytest.mark.parametrize("cutoff", [1.1])
+@pytest.mark.parametrize("precision", [np.float64, np.float32])
+@pytest.mark.parametrize("num_atoms_ligand", [1, 15])
+@pytest.mark.parametrize("num_atoms", [33, 231])
+def test_nonbonded_interaction_group_set_atom_idxs(
+    num_atoms, num_atoms_ligand, precision, cutoff, beta, rng: np.random.Generator
+):
+    box = 3.0 * np.eye(3)
+    conf = rng.uniform(0, cutoff * 10, size=(num_atoms, 3))
+    params = rng.uniform(0, 1, size=(num_atoms, 4))
+
+    ligand_idxs = rng.choice(num_atoms, size=(num_atoms_ligand,), replace=False).astype(np.int32)
+    other_idxs = np.setdiff1d(np.arange(num_atoms), ligand_idxs)
+
+    potential = NonbondedInteractionGroup(num_atoms, ligand_idxs, beta, cutoff)
+    unbound_pot = potential.unbound_impl(precision)
+
+    ref_du_dx, ref_du_dp, ref_u = unbound_pot.execute(
+        conf,
+        params,
+        box,
+    )
+
+    # Set to first particle not in ligand_idxs, should produce different values
+    unbound_pot.set_atom_idxs(other_idxs[:1])
+
+    diff_du_dx, diff_du_dp, diff_u = unbound_pot.execute(
+        conf,
+        params,
+        box,
+    )
+    assert np.any(diff_du_dx != ref_du_dx)
+    assert np.any(diff_du_dp != ref_du_dp)
+    assert not np.allclose(ref_u, diff_u)
```

Comment:
Would it be worth strengthening this to check for bitwise equality with a fresh instance initialized with `other_idxs[1:]`?

Reply:
Added in [8873ca8](https://github.com/proteneer/timemachine/pull/970/commits/8873ca8c3c70d7c0f24ca67f64fb8c12623d25f3)